fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -360,15 +360,10 @@ public void delete() {
 
             while (inboundProperties.hasNext()) {
                 final Property prop = inboundProperties.next();
-                final List<Value> newVals = new ArrayList<>();
-                final Iterator<Value> propIt = property2values.apply(prop);
-                while (propIt.hasNext()) {
-                    final Value v = propIt.next();
-                    if (!node.equals(getSession().getNodeByIdentifier(v.getString()))) {
-                        newVals.add(v);
-                        LOGGER.trace("Keeping multivalue reference property when deleting node");
-                    }
-                }
+                final List<Value> newVals = property2values.apply(prop).filter(
+                        UncheckedPredicate.uncheck(x -> !node.equals(getSession().getNodeByIdentifier(x.getString()))))
```

Comment:
Per @ajs6f's preference, shall we go with a variable name ("val"?) instead of `x`?